src/main/kotlin/mine/enums/MineCell.kt

```diff
@@ -1,7 +1,29 @@
 package mine.enums
 
+import mine.domain.MineRandomPlacer.Companion.DEFAULT_MINE_NUMBER
+
 sealed class MineCell {
-    data object MINE : MineCell()
+    private var _isOpen: Boolean = false
+    val isOpen: Boolean
+        get() = _isOpen
+
+    protected fun setOpen() {
+        _isOpen = true
+    }
```

Comment:
```suggestion
    var isOpen: Boolean = false
        protected set
```
조금 더 간단하게 표현할 수 있을 것 같아요.
아니라면 private setter를 사용해도 좋구요.